src/h5web/providers/hsds/utils.ts

```diff
@@ -26,3 +35,55 @@ export function assertHsdsDataset(
     throw new Error('Expected entity to be HSDS dataset');
   }
 }
+
+export function processHsdsBaseType(hsdsBaseType: HsdsBaseType): HDF5BaseType {
+  if (hsdsBaseType.class === HDF5TypeClass.String) {
+    return hsdsBaseType;
+  }
+
+  const { class: hsdsClass, base } = hsdsBaseType;
+
+  const regex = /H5T_[A-z]+_([A-z])(\d+)(B|L)E/u;
```

Comment:
- I don't think this works with all the types listed here: https://support.hdfgroup.org/HDF5/doc/RM/PredefDTypes.html -- it works with `H5T_STD` and `H5T_IEEE` for sure, so may as well be specific about it.
- The base is all uppercase, so the character ranges can be shortened.
- It might be simpler to just have `(BE|LE)` at the end and just store that directly on the `endianness` prop, since we don't care much about it. 🤷 

So in summary: `/H5T_(STD|IEEE)_([A-Z])(\d+)(BE|LE)/u`